### ISSUE TITLE
Embedded images and uploaded files not downloaded and linked

### ISSUE THREAD

**Issue by patbaumgartner** [kununu/slackdump2html#5]
Hi. First, I need to say that I love your addition to the slackdump.

I could not find any links to files uploaded, or images embedded in the HTML files, even though they are part of the JSON downloaded by slackdump. Would that be possible to add?

Would you also take contributions?

Best,
Patrick

**Comment by DerDodo** [comment]
Hi Patrick,
thank you for the feedback! Glad that you like it!

Under "Known issues" in the README.md you'll find that "File attachments are not supported at all." (yet).
We've created a prototype for the attachments, but our goal was to be able to upload the files to OneDrive and we didn't quickly find a clean solution for referencing these files - so we took it out of the scope in favor of custom emoji support, fixing some markup, etc.
The custom emojis are base64-encoded and added directly in the markup. Adding bigger images that way would make the file size and loading times explode.

Right now we're not actively developing more features, but we want to maintain it and fix bugs.
You're totally welcome to add this feature and create a pull request! We'll review, test, and publish it then :)

All the best,
Dodo / kununu

**Comment by patbaumgartner** [comment]
Hi Dodo,

I was thinking of downloading all the files locally (not One-Drive) and
then embedding the images with a relative path.
This might be a first step in that direction.

Best,
Patrick

**Comment by DerDodo** [comment]
Yes, this is a great first step. Especially when you can then for example just upload the HTML file and the attachment folder to a file-server.
If you need any support, write me at dominik.dopplinger@kununu.com :)
